### DISCUSSION TITLE
`wails build` does not output a JS module ?

### DISCUSSION THREAD

**Discussion by RyoTagami** [wailsapp/wails#1683] [Q&A] [answered]
Hello. I'm trying wails v2 beta42.

According to the [documentation](https://wails.io/docs/reference/cli#dev), when `wails dev` command is executed, a JS module is created in the "wailsjs" folder.

> * A JS module is generated that provides the following:
>     * Javascript wrappers of your Go methods with autogenerated JSDoc, providing code hinting
>     * TypeScript versions of your Go structs, that can be constructed and passed to your go methods

However, they are not output when the `wails build` command is executed. Is this an intentional specification?

I'm planning to automate the build, but if I only run the `wails build` command, it fails to build the front-end. (Because the "wailsjs" folder doesn't exist.)

I think that the "wailsjs" folder should not be versioned. But should it be version controlled?

**Comment by leaanthony** [comment, marked as answer]
Hi. Yes it's deliberate so there are no surprises when you make your builds. You could always use `wails generate module` if you wanted to do this. I'd personally check in that code as they are core parts of your app, but maybe that's just me 😜

**Reply by RyoTagami** [reply]
Thanks for answering!

I've never heard of the `wails generate module` command before!
I tried it with my code and it works fine💛

**Reply by leaanthony** [reply]
Hmmm... you're right. It's not in the docs! Let's sort that out.
